x-pack/plugins/cases/public/components/actions/assignees/edit_assignees_selectable.test.tsx

```diff
@@ -0,0 +1,413 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import type { AppMockRenderer } from '../../../common/mock';
+import { createAppMockRenderer } from '../../../common/mock';
+import { EditAssigneesSelectable } from './edit_assignees_selectable';
+import { basicCase } from '../../../containers/mock';
+import { waitForComponentToUpdate } from '../../../common/test_utils';
+import userEvent from '@testing-library/user-event';
+import { userProfiles, userProfilesMap } from '../../../containers/user_profiles/api.mock';
+import { act, waitFor } from '@testing-library/react';
+
+jest.mock('../../../containers/user_profiles/api');
+
+describe('EditAssigneesSelectable', () => {
+  let appMock: AppMockRenderer;
+
+  /**
+   * Case has the following tags: Damaged Raccoon
+   * All available tags are: Damaged Raccoon, Physical Dinosaur, Wet Dingo
+   */
+  const props = {
+    selectedCases: [basicCase],
+    isLoading: false,
+    userProfiles: userProfilesMap,
+    onChangeAssignees: jest.fn(),
+    unknownUsers: [],
+  };
+
+  /**
+   * Case one has the following assignees: Damaged Raccoon
+   * Case two has the following assignees: Damaged Raccoon, Physical Dinosaur
+   * All available assignees are: Damaged Raccoon, Physical Dinosaur, Wet Dingo, Silly Hare, Convenient Orca
+   */
+  const propsMultipleCases = {
+    selectedCases: [
+      basicCase,
+      { ...basicCase, assignees: [...basicCase.assignees, { uid: userProfiles[1].uid }] },
+    ],
+    isLoading: false,
+    userProfiles: userProfilesMap,
+    unknownUsers: [],
+    onChangeAssignees: jest.fn(),
+  };
+
+  beforeAll(() => {
+    jest.useFakeTimers();
+  });
+
+  afterEach(() => {
+    jest.clearAllTimers();
+  });
+
+  afterAll(() => {
+    jest.useRealTimers();
+  });
+
+  beforeEach(() => {
+    appMock = createAppMockRenderer();
+    jest.clearAllMocks();
+  });
+
+  it('renders correctly', async () => {
+    const result = appMock.render(<EditAssigneesSelectable {...props} />);
+
+    expect(result.getByTestId('cases-actions-assignees-edit-selectable')).toBeInTheDocument();
+    expect(result.getByPlaceholderText('Search')).toBeInTheDocument();
+    expect(result.getByText('Selected: 1')).toBeInTheDocument();
+
+    for (const userProfile of userProfiles) {
+      // @ts-ignore: full name exists
+      expect(result.getByText(userProfile.user.full_name)).toBeInTheDocument();
+    }
+
+    await waitForComponentToUpdate();
+  });
+
+  it('renders the selected assignees label correctly', async () => {
+    const result = appMock.render(<EditAssigneesSelectable {...propsMultipleCases} />);
+
+    expect(result.getByText('Selected: 2')).toBeInTheDocument();
```

Comment:
What about internationalization? `i18n.SELECTED_CASES(selectedCases.length)`
When it changes the tests fail. 

(there are a few other instances of this)

Reply:
I prefer to not use internationalization in tests (and variables in general) because a) it is clearer to see what we are testing (otherwise I have to go to check the `SELECTED_CASES` variable ) and b) I want the test to fail if I change the text in case I forgot something important.